### DISCUSSION TITLE
Enabling ratelimit module for outbound emails

### DISCUSSION THREAD

**Discussion by denisfidel** [rspamd/rspamd#3622] [Q&A] [answered]
Hello!

I've been figuring this out for quite some time now, but no luck so far. 

The problem is the following. I've successfully added our organization local networks, so that emails from our IPs are treated as outbound. But as many smaller organizations use our mail servers for outgoing emails we want to enable for example ratelimit module (not just authenticated users bul also based on IP). As the documentation [here](https://rspamd.com/doc/tutorials/scanning_outbound.html)  says some checks can be enabled with setting module. But in setting [documentation](https://rspamd.com/doc/configuration/settings.html#settings-structure) there is only option to enable a group (which is not ok since then all other checks will be disabled if I enable ratelimit) or to disable a group (which is not ok, since than I cant disable e.g. graylisting because its ungrouped + i don't know how to switch from DKIM checking to signing).

 So my question is, is there a way to just say: enable full ratelimiting (and e.g. RBL) and keep using (or not using) all other modules as default, when form local nets? 

Here is how I meant to do it: 

```
outgoingLocal {
        id = "outgoingLocal";
        priority = low;
        local = yes;
        apply {
                actions {
                        # keep using all modules just add ratelimit (and e.g. RBL)
                }
        }
}
```

Would appreciate any kind of help or hint about how to achieve this :)

Kind regards,
Denis

**Comment by fatalbanana** [comment]
[RBL module](https://rspamd.com/doc/modules/rbl.html) has `exclude_private_ips` setting (though unlikely you find such addresses listed in RBLs).

Ratelimit refuses to operate on such IPs unfortunately but that's fixable.

**Reply by denisfidel** [reply]
Thank you for your answer! If its possible it would be a major feature for us because we have troubles with blacklisting. Would really appreciate it :) Can you give any info about when it could be available?

If needed I can open an issue. 

Regards,
Denis

**Reply by fatalbanana** [reply, marked as answer]
Likely possibility is to use selectors for ratelimits. This selector will return local IP addresses only:

~~~
ip:is_local.equal(true).id;ip
~~~
